src/core/xref.js

```diff
@@ -37,7 +41,35 @@ export async function run(conf, elems) {
     upgradeDB.createObjectStore("xrefs");
   });
   const cache = new IDBKeyVal(idb, "xrefs");
+  const isObject =
```

Comment:
Yeah, the logic is right, but way too convoluted :)  A better approach is to use object spread... see an example here where we do something similar:

https://github.com/w3c/respec/blob/develop/src/core/github.js#L117

So, in this case, start with a default object, and then go through each case:

```JS

const xrefDefaults = {
   url: API_URL,
   profile: "",
   specs: null,
}

// in run()
if (!conf.xref) return;

const normalizedConfig = {}; 
// get the type.. 
const type = Array.isArray(conf.xref) ? "array" : typeof conf.xref;
switch(type) {
   case "boolean": 
      // ... true
      Object.assign(normalizedConfig,  xrefDefaults);
      break;
   case "string":
      // do we have this profile? no: fallback to true, but show an error to the the user. 
      break;
   case "array":
       Object.assign(normalizedConfig, xrefDefaults, {specs: [].concat(conf.xref)});
      break;
   case "object":
         Object.assign(normalizedConfig, xrefDefaults, conf.xref);
         break;
    default: 
       // show user error - bad configuration 
       Object.assign(normalizedConfig, xrefDefaults);
}
// now apply the `normalizedConfig`... 

```

Reply:
Thanks, I will make the changes by tomorrow.